installation-manager-core/src/main/java/com/codenvy/im/license/CodenvyLicense.java

```diff
@@ -81,6 +83,36 @@ public LicenseType getLicenseType() {
         return (LicenseType)doGetFeature(TYPE);
     }
 
+    /**
+     * @return true:
+     * 1) if (EVALUATION_PRODUCT_KEY IS NOT expired) AND (actual number of users <= allowed by license)
+     * 2) if (PRODUCT_KEY            IS NOT expired) AND (actual number of users <= allowed by license)
+     * 3) if (EVALUATION_PRODUCT_KEY IS     expired) AND ((actual number of users <= MIN_NUMBER_OF_FREE_USERS) AND (number of nodes <= MIN_NUMBER_OF_FREE_SERVERS))
+     * 4) if (PRODUCT_KEY            IS     expired) AND (actual number of users <= MIN_NUMBER_OF_FREE_USERS)
+     */
+    public boolean isLicenseUsageLegal(long actualUsers, int actualServers) {
+        if (isExpired()) {
+            switch (getLicenseType()) {
+                case EVALUATION_PRODUCT_KEY:
+                default:
+                    return isFreeUsageLegal(actualUsers, actualServers);
+
+                case PRODUCT_KEY:
+                    return actualUsers <= MIN_NUMBER_OF_FREE_USERS;   // don't take into account minimal free number of servers
+            }
+        }
+
+        return actualUsers <= getNumberOfUsers();
+    }
+
+    /**
+     * @return false if (actual number of users > MIN_NUMBER_OF_FREE_USERS) OR (actual number of nodes > MIN_NUMBER_OF_FREE_SERVERS)
+     */
+    public static boolean isFreeUsageLegal(long actualUsers, int actualServers) {
+        return actualUsers <= MIN_NUMBER_OF_FREE_USERS
```

Comment:
Why MIN_NUMBER_OF_FREE_USERS? I understand this constant as maximum number of users for free usage.